### PR TITLE
Wip/6.0 re enabling tests

### PR DESCRIPTION
**Pull request by dreab8**

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/orm/test/ops/MergeManagedAndCopiesAllowedTest.java [hunk 1]
```diff
@@ -17,76 +17,74 @@
...
- *                      V                 V
- *                      1                 N
- *                      D 1------------>N E
- *
+ * 1                 1
+ * |                 |
+ * |                 |
+ * V                 V
+ * 1                 N
+ * D 1------------>N E
```
Comment (beikov) [inline hibernate-core/src/test/java/org/hibernate/orm/test/ops/MergeManagedAndCopiesAllowedTest.java:40]: I believe this formatting change was not intentional?

File: hibernate-core/src/main/java/org/hibernate/boot/query/HbmResultSetMappingDescriptor.java [hunk 2]
```diff
@@ -257,7 +260,22 @@ public NamedResultSetMappingMemento resolve(ResultSetMappingResolutionContext re
...
+				final DynamicFetchBuilderLegacy fetchBuilder = Builders.fetch(
+						descriptor.tableAlias,
+						ownerTableAlias,
+						descriptor.propertyPath
+				);
+				fetchBuilderLegacies.add( fetchBuilder );
+			}
+		}
+
+		return new NamedResultSetMappingMementoImpl( registrationName, resultMementos, fetchBuilderLegacies );
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/boot/query/HbmResultSetMappingDescriptor.java:278]: Shouldn't this be handled by `EntityResultDescriptor#resolve` where the join descriptors are resolved into fetch mementos?

File: hibernate-core/src/main/java/org/hibernate/query/sql/internal/ResultSetMappingProcessor.java [hunk 3]
```diff
@@ -322,7 +323,7 @@ private void processScalarReturn(NativeQuery.ReturnProperty typeReturn) {
 //		scalarTypes.add( typeReturn.getType() );
 	}
 
-	private void processRootReturn(NativeQuery.RootReturn rootReturn) {
+	private void processRootReturn(NativeQuery.EntityReturn rootReturn) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sql/internal/ResultSetMappingProcessor.java:326]: Maybe you should rename the method to `processEntityReturn` and also the variables that refer to such a type accordingly to `entityReturn`

File: hibernate-core/src/main/java/org/hibernate/query/sql/internal/ResultSetMappingProcessor.java [hunk 4]
```diff
@@ -341,6 +342,13 @@ private void addPersister(String alias, Map<String, String[]> propertyResult, SQ
 		entityPropertyResultMaps.put( alias, propertyResult );
 	}
 
+	private void addPersister(String alias, Map<String, String[]> propertyResult, String suffix, SQLLoadable persister) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sql/internal/ResultSetMappingProcessor.java:345]: Is this even used?

**Review by beikov — CHANGES_REQUESTED** [review]

File: hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java [hunk 5]
```diff
@@ -2114,7 +2125,7 @@ private void consumeEntityJoin(SqmEntityJoin sqmJoin, TableGroup lhsTableGroup)
 		consumeExplicitJoins( sqmJoin, tableGroupJoin.getJoinedGroup() );
 		consumeReusablePaths( sqmJoin, tableGroupJoin.getJoinedGroup() );
 
-		// add any additional join restrictions
+		// add any additional join restrictionsHbmResultSetMappingDescriptor.java
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java:2128]: Typo

```suggestion
		// add any additional join restrictions
```

**Comment by dreab8** [comment]
Thanks @beikov for the review and great help!

**Pull request merged**